evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -1938,3 +2457,344 @@ def test_partial_dependence_jupyter_graph_check(
+    X, y = X_y_binary
+    X = pd.DataFrame(X)
+    X.columns = X.columns.astype(str)
+
+    es = ft.EntitySet()
+    es = es.add_dataframe(
+        dataframe_name="X",
+        dataframe=X,
+        index="index",
+        make_index=True,
+    )
+    X_fm, features = ft.dfs(
+        entityset=es,
+        target_dataframe_name="X",
+        trans_primitives=["absolute"],
+    )
+
+    dfs_transformer = DFSTransformer(features=features)
+    pipeline = BinaryClassificationPipeline(
+        [dfs_transformer, "Standard Scaler", "Random Forest Classifier"],
+    )
+
+    # When fit on X, the features expected by the dfs transformer aren't present, so they're created via
+    # calculate feature matrix, and pd won't work with dfs transformer, for two reasons
+    # 1. It doesn't create a feature provenance, so only the base feature gets updated with new pd values - not any engineered features
+    # 2. Any multi input features wouldn't get created, because the other inputs wouldn't be present
+    pipeline.fit(X, y)
+    error = "Cannot use fast mode with DFS Transformer when features are unspecified or not all present in X."
+    with pytest.raises(
+        PartialDependenceError,
+        match=error,
+    ):
+        partial_dependence(
+            pipeline,
+            X,
+            features=1,
+            grid_resolution=5,
+            fast_mode=True,
+            X_train=X,
+            y_train=y,
+        )
+
+    # If we pass the feature matrix into the same pipeline, though, DFS transformer will be no op, so pd should match
+    pipeline = pipeline.clone()
+    pipeline.fit(X_fm, y)
+    part_dep = partial_dependence(pipeline, X_fm, features=1, grid_resolution=5)
+    fast_part_dep = partial_dependence(
+        pipeline,
+        X_fm,
+        features=1,
+        grid_resolution=5,
+        fast_mode=True,
+        X_train=X_fm,
+        y_train=y,
+    )
+    pd.testing.assert_frame_equal(part_dep, fast_part_dep)
+
+
+@pytest.mark.parametrize("problem_type", [ProblemTypes.BINARY, ProblemTypes.REGRESSION])
+def test_partial_dependence_fast_mode_ensemble_pipeline_blocked(
+    problem_type,
+    X_y_binary,
+    X_y_regression,
+):
+    if problem_type == ProblemTypes.BINARY:
+        X, y = X_y_binary
+        input_pipelines = [
+            BinaryClassificationPipeline(
+                ["Standard Scaler", "Random Forest Classifier"],
+            ),
+            BinaryClassificationPipeline(["Standard Scaler", "Elastic Net Classifier"]),
+        ]
+    else:
+        X, y = X_y_regression
+        input_pipelines = [
+            RegressionPipeline(["Random Forest Regressor"]),
+            RegressionPipeline(["Elastic Net Regressor"]),
+        ]
+    pipeline = _make_stacked_ensemble_pipeline(
+        input_pipelines=input_pipelines,
+        problem_type=problem_type,
+    )
+    pipeline.fit(X, y)
+
+    error = re.escape(
+        f"cannot run partial dependence fast mode",
+    )
```

Comment:
This might be something w.r.t regex I'm unfamiliar with, but why do we need `re.escape` here? Does keeping this as a regular string fail?

Reply:
Nope, I think this was just an accidental holdover from when I tried to include the full error message with the component name and the extra quotation marks and whatnot weren't playing nice. I'll remove!